rust/migrate-wicked/src/migrate.rs

```diff
@@ -15,18 +14,37 @@ impl WickedAdapter {
 
 impl Adapter for WickedAdapter {
     fn read(&self) -> Result<model::NetworkState, Box<dyn std::error::Error>> {
-        task::block_in_place(|| {
-            Handle::current().block_on(async {
-                let interfaces = wicked_read(self.paths.clone())?;
-                let mut state = NetworkState::new(vec![], vec![]);
-
-                for interface in interfaces {
-                    let conn: model::Connection = interface.into();
-                    state.add_connection(conn)?;
+        let interfaces = wicked_read(self.paths.clone())?;
+
+        if !MIGRATION_SETTINGS.get().unwrap().continue_migration && interfaces.warning.is_some() {
+            return Err(interfaces.warning.unwrap().into());
+        }
+
+        let mut state = NetworkState::new(vec![], vec![]);
+
+        for interface in interfaces.interfaces {
+            let connection_result = interface.to_connection()?;
+            if !connection_result.warnings.is_empty()
```

Comment:
I think I would structure the code like:
```
if !vec.empty() {
   if args.opt1  {
     do... a;
   }
   if args.opt2 {
    do... b;
   }
}
````
wdyt, I don't have hard feelings here :-)